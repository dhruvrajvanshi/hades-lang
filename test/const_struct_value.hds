
struct F {
    x: *u8
}

struct VTable {
    run: fn() -> void
}

fn i32Run(): void {
    puts(c"i32Run")
}

const i32VTable: VTable = VTable(i32Run)

fn f64Run(): void {
    puts(c"f64Run")
}
const f64VTable: VTable = VTable(f64Run)

fn main(): void {
    run(i32VTable)
    run(f64VTable)
}


fn run(vtable: VTable): void {
    vtable.run()
}

extern fn puts(*cchar): void = puts
