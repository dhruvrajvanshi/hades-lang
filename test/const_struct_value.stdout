i32Run
f64Run